src/base/materialized.ts

```diff
@@ -230,60 +236,203 @@ export class MaterializedChatMessage implements IMaterializedNode {
 		return tokenizer.countMessageTokens({ ...this.toChatMessage(), content: '' });
 	});
 
-	private readonly _text = once(() => {
-		let result = '';
+	private readonly _text = once((): (string | MaterializedChatMesageImage)[] => {
+		let result: (string | MaterializedChatMesageImage)[] = [''];
 		for (const { text, isTextSibling } of textChunks(this)) {
+			if (text instanceof MaterializedChatMesageImage) {
+				result.push(text);
+				for (const child of text.children) {
```

Comment:
I think we should enforce, for the moment, that an image element doesn't have any children. You _might_ be able to do this in the JSX props by setting `children: undefined | [];`, but I'm not sure if that'll work.